File: backend/src/main/scala/microservices/blog/api/GetBlogPostPlanner.scala
// GetBlogPostPlanner：博客域获取博客文章入口。

package com.typesafe.travel.blog.domain

import cats.effect.IO
import com.typesafe.travel.api.routes.ConnectionApiPlan
import com.typesafe.travel.persistence.blog.BlogPlannerPlainSql

import java.sql.Connection

object GetBlogPostPlanner extends ConnectionApiPlan[BlogPostByIdPlannerRequest, BlogPostResponse]:
  override val name: String = "GetBlogPostPlanner"
  override def plan(input: BlogPostByIdPlannerRequest, connection: Connection): IO[BlogPostResponse] =
    BlogPlannerPlainSql.findPost(connection, input.postId, input.userId)

File: frontend/src/microservices/blog/api/GetBlogPostPlanner.ts
// 本文件定义 GetBlogPostPlanner，负责博客域对应接口入口。

import type { BlogPostResponse } from '@/microservices/blog/objects/BlogPostResponse'
import { executeJsonApiRequest } from '@/shared-kernel/api/ApiTransport'

export const getBlogPost = (postId: string, userId?: string): Promise<BlogPostResponse> =>
  executeJsonApiRequest('/GetBlogPostPlanner', 'POST', { postId, userId })
